third_party/move/move-compiler-v2/tests/checking/naming/generics_shadowing_invalid.exp

```diff
@@ -1,24 +1,24 @@
 
 Diagnostics:
-error: expected `M::S` but found `S`
+error: cannot adapt `S` to annotated type `S`
   ┌─ tests/checking/naming/generics_shadowing_invalid.move:7:10
   │
 7 │         (s1: Self::S);
   │          ^^
 
-error: expected `S` but found `M::S`
+error: cannot adapt `S` to annotated type `S`
   ┌─ tests/checking/naming/generics_shadowing_invalid.move:8:20
   │
 8 │         let s: S = S {};
   │                    ^^^^
 
-error: invalid call of `M::bar`: expected `M::S` but found `S` for argument 1
+error: cannot pass `S` to a function which expects argument of type `S`
```

Comment:
This is no good.  File a new issue if we don't fix it here.

Reply:
Fixed, the display is now aware of the name clash and does not shorten the name if so.